src/Ilios/CoreBundle/Service/Directory.php

```diff
@@ -57,10 +57,17 @@ public function findByCampusIds(array $campusIds)
         $filterTerms = array_map(function ($campusId) use ($ldapCampusIdProperty) {
             return "({$ldapCampusIdProperty}={$campusId})";
         }, $campusIds);
-        $filterTermsString = implode($filterTerms, '');
-        $filter = "(|{$filterTermsString})";
 
-        $users = $this->ldapManager->search($filter);
+        $users = [];
+
+        //Split into groups of 50 to avoid LDAP query length limits
+        foreach (array_chunk($filterTerms, 50) as $terms) {
```

Comment:
perhaps a foolish question, but: is chunking this down to bites of 50 returns at a time going to run the risk of triggering any DOS block from the directory, when total request returns are of 1000+? It would be dreadful to solve this problem only to instigate one that would require policy or policy discussion remedies rather than technical ones.

Also, I realize the lunacy of this hypothetical. But I also am aware of the current status of ITS Governance.

Reply:
50 was what ITS suggested . We will need to put attention, but we have documentation that this is a good number.